spec/controllers/idv/in_person/usps_locations_controller_spec.rb

```diff
@@ -25,53 +38,115 @@
     stub_sign_in(user) if user
     allow(IdentityConfig.store).to receive(:in_person_proofing_enabled).
       and_return(in_person_proofing_enabled)
+    allow(IdentityConfig.store).to receive(:arcgis_search_enabled).
+      and_return(arcgis_search_enabled)
     allow(controller).to receive(:current_sp).and_return(sp)
   end
 
   describe '#index' do
     let(:proofer) { double('Proofer') }
     let(:locations) do
+      [
+        { address: '3118 WASHINGTON BLVD',
+          city: 'ARLINGTON',
+          distance: '6.02 mi',
+          name: 'ARLINGTON',
+          phone: '703-993-0072',
+          saturday_hours: '9:00 AM - 1:00 PM',
+          state: 'VA',
+          sunday_hours: 'Closed',
+          weekday_hours: '9:00 AM - 5:00 PM',
+          zip_code_4: '9998',
+          zip_code_5: '22201' },
+        { address: '4005 WISCONSIN AVE NW',
+          city: 'WASHINGTON',
+          distance: '6.59 mi',
+          name: 'FRIENDSHIP',
+          phone: '202-842-3332',
+          saturday_hours: '8:00 AM - 4:00 PM',
+          state: 'DC',
+          sunday_hours: '10:00 AM - 4:00 PM',
+          weekday_hours: '8:00 AM - 6:00 PM',
+          zip_code_4: '9997',
+          zip_code_5: '20016' },
+        { address: '6900 WISCONSIN AVE STE 100',
+          city: 'CHEVY CHASE',
+          distance: '8.99 mi',
+          name: 'BETHESDA',
+          phone: '301-941-2670',
+          saturday_hours: '9:00 AM - 4:00 PM',
+          state: 'MD',
+          sunday_hours: 'Closed',
+          weekday_hours: '9:00 AM - 5:00 PM',
+          zip_code_4: '9996',
+          zip_code_5: '20815' },
+      ]
+    end
+    let(:pilot_locations) do
       [
         { name: 'Location 1' },
         { name: 'Location 2' },
         { name: 'Location 3' },
         { name: 'Location 4' },
       ]
     end
-    subject(:response) { get :index }
+    subject(:response) do
+      post :index, params: { address: { street_address: '1600 Pennsylvania Ave',
+                                        city: 'Washington',
+                                        state: 'DC', zip_code: '20500' } }
+    end
 
     before do
       allow(UspsInPersonProofing::Proofer).to receive(:new).and_return(proofer)
     end
 
-    context 'with successful fetch' do
-      before do
-        allow(proofer).to receive(:request_pilot_facilities).and_return(locations)
+    context 'with arcgis search enabled' do
+      context 'with successful fetch' do
+        before do
+          allow(proofer).to receive(:request_facilities).with(address).and_return(locations)
```

Comment:
zach agrees!

Reply:
I implemented this change in the controller.  I haven't changed the test setup, but can do so tomorrow.  I'm logging off for the day, but I am also considering mocking the proofer.  Sheldon suggested this refactor, and noted this method in the `enrollment_helper.rb` as potential inspiration.
```
      def usps_proofer
        if IdentityConfig.store.usps_mock_fallback
          UspsInPersonProofing::Mock::Proofer.new
        else
          UspsInPersonProofing::Proofer.new
        end
      end
```

Reply:
zach also likes mocking the proofer, esp useful for lower envs

that said, I'm wondering if it would be ok to use "real" arcgis in a lower env, like we use real SMS and real voice so people can acually use the app

Reply:
Good question.  I'll bring this up during the next PO search check-in meeting.